packages/manager/modules/web-paas/src/details/service/change-offer/change-offer.routing.js

```diff
@@ -0,0 +1,74 @@
+import { compact, find, map, set } from 'lodash';
+import { commonResolves } from '../../../add/add.utils';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('web-paas.dashboard.service.change-offer', {
+    url: '/change-offer',
+    views: {
+      projectDetailsView: 'webPaasChangeOfferComponent',
+    },
+    params: {
+      cpu: null,
+    },
+    resolve: {
+      ...commonResolves,
+      cpu: /* @ngInject */ ($transition$) => $transition$.params().cpu,
+      selectedProject: /* @ngInject */ (WebPaas, projectId) =>
+        WebPaas.getProjectDetails(projectId),
+      availablePlans: /* @ngInject */ (selectedProject, WebPaas, cpu) =>
+        WebPaas.getUpgradeOffers(selectedProject.serviceId).then((data) => {
+          if (cpu) {
+            return compact(
+              map(data, (plan) =>
+                plan.productName === selectedProject.offer.split('-')[0]
+                  ? plan
+                  : null,
```

Comment:
you want to return null or empty array?

Reply:
since i am using `compact`, `null` is okay.